### ISSUE TITLE
JAVA_HOME Issues with sbt and "runMain UDFDemo"

### ISSUE THREAD

**Issue by brianmurph** [Snowflake-Labs/sfquickstarts#148]
The operation sbt "runMain UDFDemo" would not complete because a compatible SDK could not be found and JAVA_HOME was not correctly defined.

The instructions provided did not specify the download of any Java SDK and only the JRE was provided through the documented link.

Even when I installed jdk-8u_161-windows-x64.exe and jdk-8u_161-windows-i586.exe (tried both combinations) and set JAVA_HOME (and windows path to %JAVA_HOME%\bin) the problem remained.

I imagine that there is some easy solution (but it eludes me right now), as all of the other sbt options worked.

**Comment by brianmurph** [comment]
![Screenshot from 2021-10-11 15-56-18](https://user-images.githubusercontent.com/32259660/136737136-b3208525-9c80-494c-a979-d00613f1db17.png)
![Screenshot from 2021-10-11 16-13-24](https://user-images.githubusercontent.com/32259660/136737143-17a238be-7252-4e42-8808-71c955269cd7.png)

Here are 2 screenshots describing the issue with the UDF compilations (both on different Windows installs), one 32 bit and the other 64 bit.

**Comment by sfc-gh-jwest** [comment]
@brianmurph what do you get returned, when you run:  java --version?

**Comment by brianmurph** [comment]
Hi Jason, thanks for getting back to me.

I have attached some of the Java environment information (including the version).


![Java_Properties](https://user-images.githubusercontent.com/32259660/138184735-9c52bc81-be78-460e-9bcd-542f861b431f.png)
![Java_Arguments](https://user-images.githubusercontent.com/32259660/138184737-35cc1b2b-a010-48c0-9841-b0037b85b755.png)
![Java_Version](https://user-images.githubusercontent.com/32259660/138184740-c52fd203-78dc-44d8-9402-287912f76318.png)
![Java_Home](https://user-images.githubusercontent.com/32259660/138184742-4e3935f6-a71a-4ccd-be91-4cf56c9b3091.png)

**Comment by sfc-gh-jwest** [comment]
@brianmurph try moving the java_home/bin to the front of your path.  Also try entering the full path (in quotes or with \ to escape the spaces) to the front of your PATH and drop the %java_home%

**Comment by brianmurph** [comment]
Hi Jason,

I will uninstall the JDK and try installing to another directory (ie. Java_SDK), so there are no blank spaces in the Path or JAVA_HOME. Thanks for the tip.

I will update you on the outcome.

**Comment by brianmurph** [comment]
Hi Jason,

After trying many combinations of Windows PATH and JAVA_HOME, I gave up on that idea.

I exported the JAVA_HOME (i.e. export JAVA_HOME=/c/Java_JDK) and the PATH (i.e. PATH=$JAVA_HOME/bin:$PATH) in MINGW64 and that worked. In the interim I also upgraded to Java 1.8.0_311, which also worked fine.

I now have a .bashrc with these settings and am happy with that (I really should have remembered my Linux training).

Anyway, thanks for your assistance - I can now compile the UDFs.
